### DISCUSSION TITLE
too many concurrency async calls using promise.all kills nodejs

### DISCUSSION THREAD

**Discussion by weidezhang** [nodejs/help#4696] [Q&A]
hi I'm new to node but a test code I write kind of confuse me with the nodejs behavior and eventually node killed without any specific reason. My code is like the following: 

```
   const array2 = await Promise.all(
         files.map(async (file) => {
                await Promise.all(
                    otherfiles.map(async (otherfile) => {
                         await some_async_op();  
                   }
               })
        );
           await some_other_async_op(); 
       })
 );
```
when the files is an array of about 100 elements, otherfiles is an array of about 10 elements. it seems node starts executing at the begining but it never completes. And eventually it killed itself. Any idea why it is the case and how shall i make it work ?  Thanks a lot

**Comment by bnoordhuis** [comment]
Hard to say. Can you post a complete example? Your code snippet isn't syntactically valid.

**Comment by aduh95** [comment]
What is the exit code you are seeing?
